From ced0dbcd7c24aec4da31da9ebbfc588736d7de33 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 10 Aug 2012 17:10:09 +0800
Subject: [PATCH 326/486] ENGR00220081 Power supply to camera is 2.5V which
 should be 2.7v

video0/1 both opened will lower the power supply to ov sensor

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/V4l2CapDeviceBase.cpp | 5 ++---
 mx6/libcamera/V4l2CsiDevice.cpp     | 5 ++---
 mx6/libcamera/V4l2UVCDevice.cpp     | 9 ++-------
 3 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/mx6/libcamera/V4l2CapDeviceBase.cpp b/mx6/libcamera/V4l2CapDeviceBase.cpp
index 26e509e..f563e62 100755
--- a/mx6/libcamera/V4l2CapDeviceBase.cpp
+++ b/mx6/libcamera/V4l2CapDeviceBase.cpp
@@ -274,10 +274,9 @@ namespace android{
                             strcpy(mCaptureDeviceName, dev_node);
                             break;
                         }
-                    } else {
-                        close(fd);
-                        fd = 0;
                     }
+                    close(fd);
+                    fd = 0;
                 }
                 closedir(v4l_dir);
             }
diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 6ddd2cb..f3f4d0d 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -100,10 +100,9 @@ namespace android{
                         CAMERA_LOG_INFO("device name is %s", mCaptureDeviceName);
                         CAMERA_LOG_INFO("sensor name is %s", mInitalDeviceName);
                         break;
-                    } else{
-                        close(fd);
-                        fd = 0;
                     }
+                    close(fd);
+                    fd = 0;
                 }
                 closedir(v4l_dir);
             }
diff --git a/mx6/libcamera/V4l2UVCDevice.cpp b/mx6/libcamera/V4l2UVCDevice.cpp
index 11dd0d5..1c3eb99 100755
--- a/mx6/libcamera/V4l2UVCDevice.cpp
+++ b/mx6/libcamera/V4l2UVCDevice.cpp
@@ -116,14 +116,9 @@ CAPTURE_DEVICE_RET V4l2UVCDevice::V4l2Open(int cameraId)
                             strcpy(mCaptureDeviceName, dev_node);
                             break;
                         }
-                        else {
-                            close(fd);
-                            fd = 0;
-                        }
-                    } else {
-                        close(fd);
-                        fd = 0;
                     }
+                    close(fd);
+                    fd = 0;
                 }
                 closedir(v4l_dir);
             }
-- 
1.8.0

